core/store/src/trie/shard_tries.rs

```diff
@@ -13,18 +13,28 @@ use near_primitives::types::{
 
 use crate::db::{DBCol, DBOp, DBTransaction};
 use crate::trie::trie_storage::{TrieCache, TrieCachingStorage};
-use crate::trie::TrieRefcountChange;
+use crate::trie::{TrieRefcountChange, POISONED_LOCK_ERR};
 use crate::{StorageError, Store, StoreUpdate, Trie, TrieChanges, TrieUpdate};
 
-#[derive(Clone)]
-pub struct ShardTries {
-    pub(crate) store: Arc<Store>,
+struct ShardTriesInner {
+    store: Arc<Store>,
     /// Cache reserved for client actor to use
-    pub(crate) caches: Arc<HashMap<ShardUId, TrieCache>>,
+    caches: RwLock<HashMap<ShardUId, TrieCache>>,
     /// Cache for readers.
-    pub(crate) view_caches: Arc<HashMap<ShardUId, TrieCache>>,
```

Comment:
:+1: for reducing visibility! It's so much easier to understand the code when you know that random things don't poke internals of structs!